pkg/runtime/session.go

```diff
@@ -42,8 +42,17 @@ func (c *serviceController) NewSession(
 		Region:              aws.String(string(region)),
 		STSRegionalEndpoint: endpoints.RegionalSTSEndpoint,
 	}
-	if endpointURL != nil {
-		awsCfg.Endpoint = endpointURL
+
+	if *endpointURL != "" {
+		endpointServiceResolver := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+			if service == c.ServiceEndpointsID {
+				return endpoints.ResolvedEndpoint{
+					URL: *endpointURL,
```

Comment:
There is also a "SigningRegion" field. Do you know if that is applicable, or what that's used for?

Reply:
There are various "signing" fields. Basing this code off of the old operators for sagemaker it was not specified. https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#ResolvedEndpoint. I believe it has to do with https://docs.aws.amazon.com/general/latest/gr/signing_aws_api_requests.html, but I don't think it is really necessary.

Reply:
Yeah, Googling the subject seems to show few results. I imagine it defaults to a sane value.